src/jit/codegencommon.cpp

```diff
@@ -7531,11 +7588,17 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper /*= CORINFO_HELP_PROF_FC
     // Need to save on to the stack level, since the helper call will pop the argument
     unsigned saveStackLvl2 = genStackLevel;
 
-#if defined(_TARGET_AMD64_) && !defined(UNIX_AMD64_ABI) // No profiling for System V systems yet.
+#if defined(_TARGET_AMD64_)
 
+#if !defined(UNIX_AMD64_ABI)
     // Since the method needs to make a profiler callback, it should have out-going arg space allocated.
     noway_assert(compiler->lvaOutgoingArgSpaceVar != BAD_VAR_NUM);
     noway_assert(compiler->lvaOutgoingArgSpaceSize >= (4 * REGSIZE_BYTES));
+#endif
+
+#if defined(UNIX_AMD64_ABI)
+    regSet.AddMaskVars(REG_R10 | REG_R11 | REG_R12 | REG_R13);
+#endif
```

Comment:
Note that this method to generate Enter callback is called after register allocation and in fact after OS prolog is generated.  OS prolog would also push any callee saved regs if required.  Isn't this late to add these regs to masks?